--- a/lfs/init.d/mountvirtfs
+++ b/lfs/init.d/mountvirtfs
@@ -71,6 +71,12 @@
           if ! mountpoint /dev >/dev/null; then
              log_info_msg2 " ${INFO}/dev"
              mount -o mode=0755,nosuid /dev  || failed=1
           fi
+
+          if [ -d /sys/firmware/efi/efivars ] && \
+             ! mountpoint /sys/firmware/efi/efivars >/dev/null; then
+             log_info_msg2 " ${INFO}efivarfs"
+             mount -t efivarfs efivarfs /sys/firmware/efi/efivars || failed=1
+          fi

           mkdir -p /dev/shm
           log_info_msg2 " ${INFO}/dev/shm"
