diff -Naur glycin-2.0.2.orig/glycin-loaders/meson.build glycin-2.0.2/glycin-loaders/meson.build
--- glycin-2.0.2.orig/glycin-loaders/meson.build	2025-10-02 15:00:41.059770451 +0200
+++ glycin-2.0.2/glycin-loaders/meson.build	2025-10-02 15:44:57.299972860 +0200
@@ -46,11 +46,17 @@
 
 manifest = meson.project_source_root() / 'Cargo.toml'
 
+rust_target = meson.is_cross_build() ? meson.get_cross_property('rust_target', '') : ''
+
 cargo_options = [
   '--profile', cargo_profile,
   '--manifest-path', manifest,
 ]
 
+if rust_target != ''
+  cargo_options += ['--target', rust_target]
+endif
+
 git_version = ''
 
 if cargo_profile != 'release'
@@ -103,7 +109,9 @@
     output: loader,
     command: [
       'cp',
-      cargo_target_dir / rust_target / loader,
+      (rust_target == ''
+        ? (cargo_target_dir / cargo_profile / loader)
+        : (cargo_target_dir / rust_target / cargo_profile / loader)),
       '@OUTPUT@',
     ],
   )
diff -Naur glycin-2.0.2.orig/libglycin/meson.build glycin-2.0.2/libglycin/meson.build
--- glycin-2.0.2.orig/libglycin/meson.build	2025-10-02 15:00:41.060875000 +0200
+++ glycin-2.0.2/libglycin/meson.build	2025-10-02 15:04:58.536987503 +0200
@@ -13,12 +13,18 @@
 gnome = import('gnome')
 manifest = meson.project_source_root() / 'Cargo.toml'
 
+rust_target = meson.is_cross_build() ? meson.get_cross_property('rust_target', '') : ''
+
 cargo_options = [
     '--target-dir', cargo_target_dir,
     '--profile', cargo_profile,
     '--manifest-path', manifest,
 ]
 
+if rust_target != ''
+  cargo_options += ['--target', rust_target]
+endif
+
 cargo_env_module = cargo_env
 
 packages = []
@@ -70,7 +76,9 @@
         endif
     endif
 
-    cargo_output = cargo_target_dir / rust_target / f'libglycin@suffix@.so'.replace('-', '_')
+    cargo_output = (rust_target == ''
+        ? (cargo_target_dir / cargo_profile / f'libglycin@suffix@.so'.replace('-', '_'))
+        : (cargo_target_dir / rust_target / cargo_profile / f'libglycin@suffix@.so'.replace('-', '_')))
 
     # Build library with Rust
     libglycin = custom_target(
