diff -rupN libidn.orig/Makefile.am libidn/Makefile.am
--- libidn.orig/Makefile.am	2018-10-16 20:36:07.230954718 +0200
+++ libidn/Makefile.am	2018-10-16 20:38:19.038641175 +0200
@@ -16,8 +16,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-SUBDIRS = gl lib/gl lib po src doc gltests lib/gltests tests examples	\
-	java csharp
+SUBDIRS = gl lib/gl lib
 
 EXTRA_DIST = cfg.mk maint.mk .clcopying
 EXTRA_DIST += COPYING.LESSERv2 COPYING.LESSERv3 COPYINGv2 COPYINGv3
